swatch-picker/open-swatch-picker.js

```diff
@@ -0,0 +1,2204 @@
+        const b = Math.round(srgb.blue);
+        return `#${r.toString(16).padStart(2, '0')}${g.toString(16).padStart(2, '0')}${b.toString(16).padStart(2, '0')}`;
+    }
+
+    oklchToRgb(oklch) {
+        const oklchOb = oklch.match(/oklch\(([^)]+)\)/)[1].replaceAll(' ', ',').split(',').map(Number);
+        const oklab = this.oklchToOklab(oklchOb);
+        const xyzD65 = this.oklabToXyzD65(oklab);
+        const linearRgb = this.xyzD65ToLinearRgb(xyzD65);
+        const srgb = this.linearRgbToSrgb(linearRgb);
+
+        const r = Math.round(srgb.red);
+        const g = Math.round(srgb.green);
+        const b = Math.round(srgb.blue);
+        return `rgb(${r}, ${g}, ${b})`;
+    }
+
+    setAllValues(oklch, variableName) {
+        this.hexValue = this.oklchToHex(oklch);
+        this.rgbValue = this.oklchToRgb(oklch);
+
+        if( this.outputType === 'hex' ) {
+            this.value = this.hexValue;
+        } else if( this.outputType === 'rgb' ) {
+            this.value = this.rgbValue;
+        } else {
+            this.value = variableName;
+        }
+    }
+
+
+    onValueChange() {
+        this.updateLabel();
+        this.dialog.close();
+    }
+
+    updateLabel() {
+        const slot = this.shadowRoot.querySelector('slot[name=input]');
+        const assignedNodes = slot.assignedNodes();
+        
+        if (assignedNodes.length > 0 && !assignedNodes[0].hasAttribute('result-container')) {
+            const inputElement = assignedNodes[0];
+            inputElement.textContent = this.value;
+        } else {
+            assignedNodes.forEach(node => node.remove());
+
+            const container = document.createElement('div');
+            container.setAttribute('result-container', '');
+
+            const output = document.createElement('span');
+            output.style.display = 'flex';
+            output.style.alignItems = 'center';
+            output.style.gap = '0.5rem';
+
+            output.innerHTML = `
+                <span style="width: 20px; height: 20px; border-radius: 3px; display: inline-block; background: ${this.value}"></span>
+            `;
+
+            if(this.showOutputLabel === 'true') output.innerHTML += `<span>${this.value}</span>`
+
+            container.appendChild(output);
+            this.appendChild(container);
+            container.setAttribute('slot', 'input');
+        }
+    }
+
+    connectedCallback() {
+        this.shadowRoot.adoptedStyleSheets = [styles]
+        this.shadowRoot.replaceChildren(template.content.cloneNode(true))
+        
+        this.dialog = this.shadowRoot.querySelector("dialog");
+
+        const slot = this.shadowRoot.querySelector('slot[name=button]');
+        const assignedNodes = slot.assignedNodes();
+        let buttonElement;
+        if( assignedNodes.length > 0 ) {
+            buttonElement = assignedNodes[0];
+            buttonElement.addEventListener("click", () => {
+                this.dialog.showModal();
+            });
```

Comment:
This event listener can be bound to either ShadowRoot or Host, and be a delegated listener, rather than binding directly to the button. Either that or use `commandfor` on the button pointing to the dialog.